TakingITGlobal/commit2act
# [TIG-793] Celebration effect for signup container

## Ticket(s)

* [TIG-793](https://app.clickup.com/t/9009201449/TIG-793) - Focused sub-task
* [TIG-770](https://app.clickup.com/t/9009201449/TIG-770) - Parent task
* [TIG-798](https://app.clickup.com/t/9009201449/TIG-798) - Additional bug

## Description

This PR wraps up the work left for [TIG-770](https://app.clickup.com/t/9009201449/TIG-770), including the celebration background-image effect on the green gradient variant of `Paper` and simplifying the `variant` styling for `gradient` type. 

There is also a fix here for `Typography` elements under `Paper` (with the `gradient` `variant`) to inherit dark colour by default - which addresses [TIG-798](https://app.clickup.com/t/9009201449/TIG-798)].

## Testing

1. Pull down and install the branch (`npm install`)
2. Run tests locally (`npm run test:basic`)
3. Build Storybook locally (`npm run storybook`)
4. Visit Components > Paper and confirm both "Gradient" and "Gradient With Effect" stories render 
5. Visit Pages > Sign Up > Step 3 and confirm the text renders in a dark colour and centered in the page
6. Confirm the green gradient background is rendered on the page
7. Visit Pages > Sign Up > Step 13 and confirm the celebration background effect renders
8. Shut down Storybook and run the dev app locally (`npm run start:dev`)
9. Visit `/onboard/details` and confirm the text and Next/Back buttons render in a dark colour (ignore the fields for now)
10. Step through each onboard page, confirming the gradient background is shown and text is black (except mocked fields)
11. Confirm the final step shows the celebration effect background image
12. Navigate to the app dashboard and confirm the stat tiles ("CO2 Saved This Week", "Total CO2 Saved", ...) have the numbers visible and matching live app's styles

## Screenshots

**`Paper` story with gradient variant rendered in Storybook:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/f1ea9395-402d-4056-80a1-1a21f6de88cc)

**Additional celebrate effect also rendering in Storybook:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/276a6054-06fa-4f1f-a020-5f84f089ef2e)

**Gradient background rendered within the app locally:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/4d21cce9-5a26-417c-bfb8-f9a0a91b65aa)

**Stat tiles on dashboard with number font styling restored:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/1a541b2d-150f-4f26-a10f-0df8dbab6a7e)
